app-drawer-layout/app-drawer-layout.html

```diff
@@ -223,46 +216,31 @@
         }
       },
 
-      resetLayout: function() {
-        this.debounce('_resetLayout', function() {
-          var drawer = this.drawer;
-          var contentContainer = this.$.contentContainer;
+      _updateLayoutStates: function() {
+        var drawer = this.drawer;
+        var contentContainer = this.$.contentContainer;
 
-          if (this.narrow || !drawer) {
+        if (this.narrow || !drawer || !drawer.getWidth) {
+          contentContainer.style.marginLeft = '';
+          contentContainer.style.marginRight = '';
+        } else {
+          var drawerWidth = drawer.getWidth();
+          if (drawer.position == 'right') {
             contentContainer.style.marginLeft = '';
-            contentContainer.style.marginRight = '';
+            contentContainer.style.marginRight = drawerWidth + 'px';
           } else {
-            var drawerWidth = drawer.getWidth();
-            if (drawer.position == 'right') {
-              contentContainer.style.marginLeft = '';
-              contentContainer.style.marginRight = drawerWidth + 'px';
-            } else {
-              contentContainer.style.marginLeft = drawerWidth + 'px';
-              contentContainer.style.marginRight = '';
-            }
-          }
-        });
-      },
-
-      _resetDrawerState: function() {
```

Comment:
@frankiefu And reset the `_narrowHasChanged` flag after setting `drawer.opened = this.openedWhenNarrow`? That sounds like it would work